### PR TITLE
Fix a man macro usage error on the rehash page.

### PR DESCRIPTION
**Pull request by eric-s-raymond**
The previous version of the NAME line was dodgy both because it was
multiline and because it listed multiple names.

Found with doclifter. While this works OK when run through man itself,
some third-party viewers and any tool that parses the markup will get
confused.

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by levitte** [comment]
Are you saying we shouldn't have multiple names anywhere? You should have a look at `doc/man3`, then...

Also, I'm curious to know the reason for `\-`

**Comment by richsalz** [comment]
This PR seems wrong.  pod2man converts the - to \- (en-dash, not minus sign).
We need and use multiple names in *many* manpages and we decided to use "openssl-rehash" for things where "rehash" is likely to conflict with an existing name (such as the shell builtin).
I am going to close this.

**Comment by eric-s-raymond** [comment]
I don't make these changes for my health.  I've been trying to clean up the man page corpus for 17 years now, and when I complain about a usage it's because I've seen it break parsing or viewing on one or more tools.

Your stuff is in the less than 2.6% of pages I can't get a clean XML-DocBook lift on without patching,  Do you want it to stay there?

**Comment by levitte** [comment]
I had a look at the produced rehash.1 and at doclifter, and it seems that the actual beef is about the NAME content being more than one line, i.e. doclifter expects one line to be complete.

This is what the relevant part in rehash.1 looks like when freshly produced by pod2man:

```
.SH "NAME"
openssl\-c_rehash, openssl\-rehash,
c_rehash, rehash \- Create symbolic links to files named by the hash values
.SH "SYNOPSIS"
```

And sure enough, that makes doclifter sad:

```
"doc/man/man1/rehash.1", line 148: error - ill-formed NAME section ' openssl\-c_rehash, openssl\-rehash,' in doc/man/man1/rehash.1, giving up.
```

However, if I just join the two NAME lines in rehash.1, doclifter is much happier:

```
.SH "NAME"
openssl\-c_rehash, openssl\-rehash, c_rehash, rehash \- Create symbolic links to files named by the hash values
.SH "SYNOPSIS"
```

Now, looking at man(7), doclifter is right to croak on the multi-line NAME section (note *"extremely important that this format is followed"*:

> ```
>    Sections
>        Sections are started with .SH followed by the heading name.
> 
>        The only mandatory heading is NAME, which should be the  first  section
>        and  be followed on the next line by a one-line description of the pro‐
>        gram:
> 
>               .SH NAME
>               item \- description
> 
>        It is extremely important that this format is followed, and that  there
>        is  a  backslash  before  the  single dash which follows the item name.
>        This syntax is used by the mandb(8) program to  create  a  database  of
>        short  descriptions  for  the  whatis(1) and apropos(1) commands.  (See
>        lexgrog(1) for further details on the syntax of the NAME section.)
> 
>        For a list of other sections that might appear in a  manual  page,  see
>        man-pages(7).
> ```

However, it seems that for example `lexgrog` is still fine with it, so the extreme importance seems a bit arguable (the following is the output with a multi-line NAME section):

```
: ; lexgrog -w doc/man/man1/rehash.1 
doc/man/man1/rehash.1: "openssl-c_rehash - Create symbolic links to files named by the hash values"
doc/man/man1/rehash.1: "openssl-rehash - Create symbolic links to files named by the hash values"
doc/man/man1/rehash.1: "c_rehash - Create symbolic links to files named by the hash values"
doc/man/man1/rehash.1: "rehash - Create symbolic links to files named by the hash values"
```

Also, one might argue that `Pod::Man` should be more diligent in making the `NAME` section into one line...  it might be worth a hack.

So in summary:

- making one line instead of multiple seems to be necessary for at least doclifter
- having a single name before ' - ' is *not* necessary
- changing ' - ' to ' \- ' is *not* necessary

**Comment by levitte** [comment]
The produced rehash.1 I checked against was made *without* this PR, FYI

**Comment by levitte** [comment]
@eric-s-raymond, I think #6268 is the answer to your trouble.
